### PR TITLE
Fix/no no

### PR DESCRIPTION
**Pull request by BotellaA**

### CODE REVIEW

File: src/geode/mesh/builder/polygonal_surface_builder.cpp [hunk 1]
```diff
@@ -388,12 +387,13 @@ namespace geode
                 PolygonEdge edge{ polygon, e };
                 const auto edge_id = polygonal_surface_.edge_from_vertices(
                     { vertices_id[e], vertices_id[e + 1] } );
-                polygon_edges[edge_id].emplace_back( std::move( edge ) );
+                polygon_edges[edge_id.value()].emplace_back(
```
Comment (panquez) [inline src/geode/mesh/builder/polygonal_surface_builder.cpp:390]: Should we add some ASSERT when using xxx.value()?

Comment (BotellaA) [reply src/geode/mesh/builder/polygonal_surface_builder.cpp:390]: If there is no value inside an optional, accessing its value throw.

File: src/geode/mesh/core/polygonal_surface.cpp [hunk 2]
```diff
@@ -434,34 +438,39 @@ namespace geode
...
-        return {};
+        return absl::nullopt;
     }
 
     template < index_t dimension >
     bool PolygonalSurfaceBase< dimension >::is_edge_on_border(
         const PolygonEdge& polygon_edge ) const
     {
-        return polygon_adjacent( polygon_edge ) == NO_ID;
+        return !polygon_adjacent( polygon_edge ).has_value();
```
Comment (panquez) [inline src/geode/mesh/core/polygonal_surface.cpp:473]: `return !polygon_adjacent( polygon_edge )` is not enough?

File: src/geode/mesh/core/polyhedral_solid.cpp [hunk 3]
```diff
@@ -1012,22 +1018,22 @@ namespace geode
...
-        return {};
+        return absl::nullopt;
     }
 
     template < index_t dimension >
     bool PolyhedralSolid< dimension >::is_polyhedron_facet_on_border(
         const PolyhedronFacet& polyhedron_facet ) const
     {
-        return polyhedron_adjacent( polyhedron_facet ) == NO_ID;
+        return !polyhedron_adjacent( polyhedron_facet ).has_value();
```
Comment (panquez) [inline src/geode/mesh/core/polyhedral_solid.cpp:1036]: same question

**Pull request merged**